libmamba/src/filesystem/u8path.cpp

```diff
@@ -6,21 +6,37 @@
 
 #include <filesystem>
 #include <string>
+#if !defined(_WIN32)
+#include <fcntl.h>
+#include <sys/stat.h>
 
-#include "mamba/core/mamba_fs.hpp"
-#include "mamba/util/string.hpp"
+// We can use the presence of UTIME_OMIT to detect platforms that provide
+// utimensat.
+#if defined(UTIME_OMIT)
+#define USE_UTIMENSAT
+#endif
+#endif
+
+#include "mamba/filesystem/u8path.hpp"
 
-namespace fs
+namespace mamba::fs
 {
 
 #if defined(_WIN32)
     // Maintain `\` on Windows, `/` on other platforms
     std::filesystem::path normalized_separators(std::filesystem::path path)
     {
         auto native_string = path.native();
-        static constexpr auto platform_separator = L"\\";
-        static constexpr auto other_separator = L"/";
-        mamba::util::replace_all(native_string, other_separator, platform_separator);
+        static constexpr wchar_t platform_separator = L'\\';
+        static constexpr wchar_t other_separator = L'/';
+
+        auto pos = native_string.find(other_separator);
```

Comment:
Why was `util::replace_all` not good enough here? Also I would prefer this loop to be a function.

Reply:
So I am trying to get to a state of multiple internal "libraries" where there is no cyclic dependencies between  them.
The goal is to restrict what code can be used where, so that for instance "filesystem" or "util" does not start calling code form "core". These restrictions would then restrict the opportunities for spaghetti code.

Or maybe I am being too dogmatic?

Reply:
It's fine but it's not clear to me how this is related to this code, is it because otherwise there is a cyclic dependency between the filesystem and utils? If so, maybe isolating that specific replace function in it's header would solve the issue?

Reply:
@Klaim I wanted to use CMake `add_library(mamba-fs OBJECT filesystem.cpp)` to literally **forbid** including code from the rest of mamba in this library (and then do the same with `mamba-util`, `mamba-solv-cpp`...).
With more investigation, it turns out it's not so easy to do: since the headers are public, they still need to be added to libmamba to be installed, so I'm not sure it's worth the trouble.

For this particular issue, it turns out that since `L"\\"` and `L"/"` are in fact single characters, a simple call to `std::replace` will do (and is probably more performant). What do you think?

Reply:
Anyway if it's tested and `std::replace` works in that specific function, I'm fine by it.

Reply:
OK then I guess we can use replace